test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -315,14 +300,16 @@
            (is (true? (set-constants-and-verify pool-context 3)))
 
            ;; The flush should still not have completed at this point because
-           ;; the last instance from the old pool is still being borrowed
-           (is (not (realized? flush-complete)))
+           ;; the last instance from the old pool is still being borrowed.
+           ;; Wait for some arbitrary (but short) period of time to see that
+           ;; that the agent is still busy - presumably handling the flush.
+           (is (false? (await-for 50 pool-agent)))
```

Comment:
Hmm.  I guess this is sort of the same question I asked above, and I guess I'm OK with leaving it as-is if you two are.